pkg/controller/oam/applicationconfiguration/render.go

```diff
@@ -81,24 +82,34 @@ type components struct {
 	trait      ResourceRenderer
 }
 
-func (r *components) Render(ctx context.Context, ac *v1alpha2.ApplicationConfiguration) ([]Workload, error) {
-	workloads := make([]Workload, 0, len(ac.Spec.Components))
+func (r *components) Render(ctx context.Context, ac *v1alpha2.ApplicationConfiguration) ([]Workload, bool, error) {
+	workloads := make([]*Workload, 0, len(ac.Spec.Components))
 	dag := dependency.NewDAG(ac.DeepCopy())
+
 	for _, acc := range ac.Spec.Components {
 		w, err := r.renderComponent(ctx, acc, ac, dag)
 		if err != nil {
-			return nil, err
+			return nil, false, err
+		}
+
+		workloads = append(workloads, w)
+	}
+
+	hasDep := false
+	res := make([]Workload, 0, len(ac.Spec.Components))
+	for i, acc := range ac.Spec.Components {
+		ready, err := r.handleDependency(ctx, workloads[i], acc, dag)
+		if err != nil {
+			return nil, false, err
 		}
-		if w == nil { // depends on other resources. Not creating now.
+		if !ready { // Will not render this workload if dependency is not satisfied. Requeue and retry later.
+			hasDep = true
 			continue
```

Comment:
I suggest we don't continue here, but give a flag(dependency not ready), workload with this flag will not be applied, but status and scope can still be updated.

In this case, AppConfig.status will show what happening for this component instead of show nothing.